examples/ex9.cpp

```diff
@@ -440,7 +2059,18 @@ double u0_function(const Vector &x)
       case 3:
       {
          const double f = M_PI;
-         return sin(f*X(0))*sin(f*X(1));
+         return .5*(sin(f*X(0))*sin(f*X(1)) + 1.); // modified by Hennes
+      }
+      case 4:
+      {
+         double scale = 0.09;
+         double G1 = (1./sqrt(scale)) * sqrt(pow(X(0), 2.) + pow(X(1) + 0.5,2.));
+         double G2 = (1./sqrt(scale)) * sqrt(pow(X(0) - 0.5,2.) + pow(X(1), 2.));
+
+         return ((pow(X(0),2.) + pow(X(1) - 0.5,2.) <= scale) &
+                 ((X(0) <= -0.05) | (X(0) >= 0.05) | (X(1) >= 0.7))) ? 1. : 0. +
```

Comment:
Do you mean `&` and `|` or `&&` and `||` here?

Reply:
Yes, and I made it a little clearer.